project_task_code/models/project_task.py

```diff
@@ -28,3 +29,22 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        res = super(ProjectTask, self).name_search(
+            name=name,
+            args=args,
+            operator=operator,
+            limit=limit)
+        if not args:
+            args = []
+        tasks = self.env['project.task']
+        # Remove here a leading character, if name is not purely digits
+        name = name if name.isdigit() else name.lstrip('T')
```

Comment:
if we make assumptions about how data looks, better do this in its own function. Then we for example can override this an change it to '#'.

I'd suggest `_name_search_parse_code(name)` which either returns the code to be searched for or something falsy, then `name` isn't a code

Reply:
another thought about this: I think what the parsing function can best do is look at https://github.com/OCA/project/blob/12.0/project_task_code/data/task_sequence.xml#L7, see if `name` starts with the prefix and suffix configured there, and the middle is only numbers.

This is not perfect because sequence prefixes can contain some variables, but we don't care about this for now.